### DISCUSSION TITLE
Using React Instantsearch hooks, nextjs and react-redux

### DISCUSSION THREAD

**Discussion by MingrenChen** [algolia/instantsearch#5334] [Q&A]
I am new to algolia and using React Instantsearch Hooks with nextjs. Right now I am having a problem with embedding a react-redux inside <InstantSearchSSRProvider/>.

I am following the example at [hooks-next](https://github.com/algolia/react-instantsearch/tree/v6.29.0/examples/hooks-next)

I understand how the example works, but when it comes to my case, I have a react-redux around my App.
If in my instantsearch page (/pages/Search.tsx) I have 
```
  const serverState = await getServerState(<Search />);
```
I cannot dispatch state from my react-redux store.

If you know how this work, please gives me a hint and I genuinely appreciate that.

**Comment by MingrenChen** [comment]
Here's an [example](https://github.com/MingrenChen/react-instantsearch-hoos-redux) of what I am talking about

**Reply by gyto** [reply]
I have found a workaround by wrapping a only search part to `getServerState` and placing it into both page options when defined
```
const IndexPage = (...) => {
  return (
   <div>
   .. redux stuff
   <Search/>
  </div>
  )
}

const Search = (...) => <InstantSearchSSRProvider> ... </InstantSearchSSRProvider> // your search options without any redux for now :(


export async function getServerSideProps({ req }) {
	const store = initializeStore();
	const apolloClient = initializeApollo()
	const serverState = await getServerState(<Search url={url}/>);

	return {
		props: {
			url,
			serverState,
			initialApolloState: apolloClient.cache.extract(),
			initialReduxState: store.getState(),
		},
	};
}
```

I still trying to find a workaround my self because some of my products have to have a redux/zustand whatever can be used with server render. I feel like we just need to import a `_app.js` into the server state of initial Algolia search

**Reply by Haroenv** [reply]
Yes, anything which is needed for the search app needs to have its providers around `<Search`, so something like `await getServerState(<ReduxProvider value={mockvalue}><Search/></ReduxProvider>)` would be needed if that context is used

**Reply by gyto** [reply]
> Yes, anything which is needed for the search app needs to have its providers around `<Search`, so something like `await getServerState(<ReduxProvider value={mockvalue}><Search/></ReduxProvider>)` would be needed if that context is used

@Haroenv 
This is really massive to use with redux and persist. I am using it with almost 4-6 unique Providers who do different types of jobs. Is there even a way to do it by extending `_app.js` itself so we don't have to define another _app file in search of the serverSide part?

**Reply by Haroenv** [reply]
You can import the component created by _app.js if you want it to be exactly the same. Do you have a sandbox example of something which is complicated now? Thanks!
